pkg/storage/gcs/gcs.go

```diff
@@ -80,47 +77,23 @@ func (d *driver) Volumes() ([]coreapi.Volume, []coreapi.VolumeMount, error) {
 	return []coreapi.Volume{vol}, []coreapi.VolumeMount{mount}, nil
 }
 
-func (d *driver) CompleteConfiguration() error {
-	// Apply global config
-	gcfg, err := clusterconfig.Get()
-	if err != nil {
-		return fmt.Errorf("unable to get global config: %s", err)
-	}
+func (d *driver) createOrUpdatePrivateConfiguration(keyfileData string) error {
+	data := make(map[string]string)
 
-	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		cur := &coreapi.Secret{
-			TypeMeta: metaapi.TypeMeta{
-				APIVersion: coreapi.SchemeGroupVersion.String(),
-				Kind:       "Secret",
-			},
-			ObjectMeta: metaapi.ObjectMeta{
-				Name:      d.Name + "-private-configuration",
-				Namespace: d.Namespace,
-			},
-		}
-		err := sdk.Get(cur)
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return fmt.Errorf("failed to get secret %s: %s", cur.Name, err)
-			}
-		}
+	data["STORAGE_GCS_KEYFILE"] = keyfileData
 
-		if cur.StringData == nil {
-			cur.StringData = make(map[string]string)
-		}
-		cur.StringData["STORAGE_GCS_KEYFILE"] = gcfg.Storage.GCS.KeyfileData
+	return util.CreateOrUpdateSecret("image-registry", "openshift-image-registry", data)
```

Comment:
Why can't we store this in private-configuration?

Reply:
You are blindly blowing away any other keys that might be in the secret.  don't do that.

Either CreateOrUpdateSecret needs to merge the content of "data" with what's already there, or callers need to merge what is already there into "data".  The former is probably better for handling update conflicts.

Reply:
updated in next push